synapse/lib/nexus.py

```diff
@@ -366,6 +369,9 @@ async def iter(self, offs: int) -> AsyncIterator[Any]:
             maxoffs = item[0] + 1
             yield item
 
+        if tellready:
+            yield None
```

Comment:
This normally yields data shaped like is `offs, item`, and this in injecting in-band control flow data.
Maybe if we yielded `-1, {('_nexs:ready', ....)}` we'd be a bit more resilient for that inband control flow data if we ever have anything else we wanted to plumb in the future. `-1` will never be a valid tracking offset and can easily be trapped and handled by consumers.

Reply:
This is def not an in-band control flow data issue since the data being processed is distinct from the structure. I'd be concerned for something assuming it can consume and just shovel the events into a nexus log and wanted to make it explicitly break if they tried. Open to discussion, but the most important thing is to ensure 100% that whatever this is can't be written to or mistaken for a nexus log entry :)